docs/source/build_from_source.rst

```diff
@@ -0,0 +1,113 @@
+Build from Source
+=================
+
+This article describes how to build and install NNI from `source code`_.
+
+We recommend using latest setuptools:
+
+.. code-block::
+
+    python -m pip install --upgrade setuptools pip wheel
+
+.. _source code: https://github.com/microsoft/nni
+
+Development Build
+-----------------
+
+If you want to build NNI for your own use, we recommend using `development mode`_.
```

Comment:
There is also a step-by-step guide in contribution guide. We might have a cross-reference.

Reply:
I have just thought about it but can't find a proper place to insert the line.
It reads like "I just want to install NNI, why do you tell me to contribute?"
I recommend develop install here merely because it's much simpler than bdist_wheel. And develop requirements are not needed if the user does not plan to make PR.
I have put this doc right above contribution guide, so the bottom bar will say "Next: How to Contribute". Maybe that's enough?

Reply:
That's on master branch. :(